nordzy-icon-theme: init at unstable-2021-12-14

Alexnortung opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Adding icon theme for Nordzy

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Alexnortung committed:
nordzy-icon-theme: init at unstable-2021-12-14

nordzy-icon-theme: init at 5c247a4f19cf9849615631d1bf77727b945b634e

nordzy-icon-theme: etc

Fixed adding multiple themes

nordzy-icon-theme: init at unstable-2021-12-14

bobby285271 commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+in
+stdenv.mkDerivation rec {
+  pname = "nordzy-icon-theme";
+  version = "5c247a4f19cf9849615631d1bf77727b945b634e";
```
Maybe this is toooooo long for `version`?
Shall we move this to `rev` and just set `unstable-yyyy-mm-dd` or follow [RFC 107](https://github.com/NixOS/rfcs/blob/083116c0be0315977d7852b02b3f54c1fb766e62/rfcs/0107-version-normalization.md) for `version`?

bobby285271 commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+  patchPhase = ''
+    runHook prePatch
+
+    substituteInPlace install.sh \
+      --replace /bin/bash ${bash}/bin/bash
```
Shall we move this to `postPatch`?

Alexnortung commented:
@bobby285271 Seems better now :)

legendofmiracles commented:
Result of `nixpkgs-review pr 150488` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nordzy-icon-theme</li>
  </ul>
</details>

Alexnortung committed:
nordzy-icon-theme: etc

Updated platforms to only limux x86_64

legendofmiracles commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,52 @@
+  nativeBuildInputs = [ gtk3 ];
+
+  postPatch = ''
+    substituteInPlace install.sh \
+      --replace /bin/bash ${bash}/bin/bash
```
```suggestion
    patchShebangs install.sh
```

legendofmiracles commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,52 @@
+    sha256 = "Jqn5CF80xlYJ7H4qI1VEj91vcKPPoMXP5+sPs0ksiC4=";
+  };
+
+  nativeBuildInputs = [ gtk3 ];
```
Is this really needed? There's no real program code here that has to be linked against gtk3

Alexnortung replied:
Well you need `gtk3` to run `gtk-update-icon-cache` which is being called in `install.sh`. Is there another way you would do this?

dasJ replied:
Is that gtk cache even used? I thought another one was built when the system path is assembled. Maybe you can just not generate it here

bobby285271 replied:
Maybe we want to set `dontDropIconThemeCache = true;`?

legendofmiracles commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,52 @@
+    description = "A free and open source icon theme using the Nord color palette and based on WhiteSur and Numix Icon Theme";
+    homepage = "https://github.com/alvatip/Nordzy-icon";
+    license = licenses.gpl3;
+    platforms = [ "x86_64-linux" ];
```
It's just some shell code, it should run on any platform, no?

Alexnortung committed:
nordzy-icon-theme: etc

set platforms to all, and added nordzy-theme-name argument

dasJ commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,52 @@
+let
+  themes-arg-string = lib.strings.concatMapStrings (theme: "-t ${theme} ") nordzy-themes;
+in
+stdenv.mkDerivation rec {
```
Any reason you can't use stdenvNoCC?

dasJ replied:
Also why is the set recursive?

Alexnortung committed:
nordzy-icon-theme: etc

Using stdvenvNoCC. Patching away the call to gtk-update-icon-cache, thus there is no longer any need for gtk3.

bobby285271 commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,54 @@
+    description = "A free and open source icon theme using the Nord color palette and based on WhiteSur and Numix Icon Theme";
+    homepage = "https://github.com/alvatip/Nordzy-icon";
+    license = licenses.gpl3;
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.linux;
```

There is a hash mismatch on darwin, check https://github.com/NixOS/nixpkgs/issues/39308 for more details.

bobby285271 commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,54 @@
+{ stdenvNoCC
+, fetchFromGitHub
+, lib
+, bash
```
Maybe this can be removed from args?

Alexnortung replied:
Oh yeah, makes sense after the change to `patchShebangs`

Alexnortung committed:
nordzy-icon-theme: etc

Removed unnecessary input. Set platforms to linux only. Using gtk3 again and making sure the cache is not removed.

Alexnortung commented:
I have set `dontDropIconThemeCache = true;` as @bobby285271 said.

Alexnortung commented:
I think this is ready :)

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+, fetchFromGitHub
+, lib
+, gtk3
+, nordzy-theme-name ? "Nordzy"
```
```suggestion
```
Are you adding this argument because otherwise it uses the name of the derivation as the icon name (`share/icons/nordzy-icon-theme-unstable-2021-12-14`), or because you think it may useful to use a name different from the default name?

If the first, it suffices to unset the `name`environment variable before running `install.sh`, as shown below.

Alexnortung replied:
I think it is useful to set the name, which is also the functionality from the upstream. I just thought it would be easiest to use with an argument.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/icons
```
```suggestion
```
There is no need to explicitly create the destination directory. The install script already does so.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+    runHook preInstall
+
+    mkdir -p $out/share/icons
+    ./install.sh --dest $out/share/icons \
```
```suggestion
    name= ./install.sh --dest $out/share/icons \
```
Unset the `name` environment variable to run the install script. This variable is used in the script and it should be a local variable to the script. nix sets it to the name of the package.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+
+    mkdir -p $out/share/icons
+    ./install.sh --dest $out/share/icons \
+      -n ${nordzy-theme-name} \
```
```suggestion
```
It may have no need to explicitly set a different name.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+{ stdenvNoCC
+, fetchFromGitHub
+, lib
+, gtk3
```
```suggestion
, gtk3
, jdupes
```

romildo replied:
Use `jdupes` to replace duplicate files with hard links, considerable reducing the needed disk space. See below.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+  '';
+
+  nativeBuildInputs = [
+    gtk3
```
```suggestion
    gtk3
    jdupes
```

romildo replied:
`jdupes` can be used to reduce disk space.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+    mkdir -p $out/share/icons
+    ./install.sh --dest $out/share/icons \
+      -n ${nordzy-theme-name} \
+      ${themes-arg-string}
```
```suggestion
      ${lib.optionalString (nordzy-themes != []) "-t " + builtins.toString nordzy-themes}
```
See prior comments.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+    ./install.sh --dest $out/share/icons \
+      -n ${nordzy-theme-name} \
+      ${themes-arg-string}
+
```
```suggestion

    # Replace duplicate files with hardlinks to the first file in each
    # set of duplicates, reducing the installed size in about 87%
    jdupes -L -r $out/share
```

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+  dontFixup = true;
+
+  meta = with lib; {
+    description = "A free and open source icon theme using the Nord color palette and based on WhiteSur and Numix Icon Theme";
```
```suggestion
    description = "Icon theme using the Nord color palette, based on WhiteSur and Numix icon themes";
```

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+  meta = with lib; {
+    description = "A free and open source icon theme using the Nord color palette and based on WhiteSur and Numix Icon Theme";
+    homepage = "https://github.com/alvatip/Nordzy-icon";
+    license = licenses.gpl3;
```
`gpl3Only` or `gpl3Plus`?

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+    homepage = "https://github.com/alvatip/Nordzy-icon";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
+      alexnortung
+    ];
```
```suggestion
    maintainers = with maintainers; [ alexnortung ];
```

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+in
+stdenvNoCC.mkDerivation {
+  pname = "nordzy-icon-theme";
+  version = "unstable-2021-12-14";
```
```suggestion
  version = "unstable-2022-01-15";
```
Use the latest revision.

romildo commented on pkgs/data/icons/nordzy-icon-theme/default.nix:
```diff
@@ -0,0 +1,56 @@
+  src = fetchFromGitHub {
+    owner = "alvatip";
+    repo = "Nordzy-icon";
+    rev = "5c247a4f19cf9849615631d1bf77727b945b634e";
+    sha256 = "Jqn5CF80xlYJ7H4qI1VEj91vcKPPoMXP5+sPs0ksiC4=";
```
```suggestion
    rev = "ee5d01733fb69c05f5f4531105036148c47f9419";
    sha256 = "HRBPBZ3Q3ZHRcGcI1zxOI2PWO9/dkkOu9JdOdpIQxac=";
```
Use the latest revision.

Alexnortung committed:
nordzy-icon-theme: unstable-2021-12-14 -> unstable-2022-01-23

Added Jdupes and made a lot of formatting.
Co-authored-by: romildo

Alexnortung commented:
Alright @romildo I fixed all the changes you requested
